src/main/java/in/org/projecteka/hiu/patient/PatientService.java

```diff
@@ -33,4 +49,38 @@
                                     return patientRep.toPatient();
                                 }));
     }
+
+    public Mono<Patient> findPatientWith(String id) {
+        return getFromCache(id, () ->
+        {
+            var cmSuffix = getCmSuffix(id);
+            var request = getFindPatientRequest(id);
+            return centralRegistry.token()
+                    .flatMap(token -> scheduleThis(gatewayServiceClient.findPatientWith(request, token, cmSuffix))
```

Comment:
I feel we can easily push the token fetching to the GatewayServiceClient now. Does not need to be a concern of the PatientService or any other Service. HIU is only talking to Gateway now.

Reply:
done